tags: focus management of removed tags. Improve docs (#1820)
agriculturegovau/agds-next: packages/react/src/tags/docs/overview.mdx

## Removable tags

Tags can be removed from a list when you pass a function to the `onRemove` prop.

When a user removes a tag, their focus is automatically moved to the previous tag in the list. When the last tag is removed, focus is not moved, and, depending on the context, may need to be manually moved in `onRemove` to help the user continue their task.

> Screen reader users need to be informed when a tag has been removed. Depending on the context, this may be done explicitly, like in the example below, or more commonly implicitly by announcing the resultant changes to the page (like in the [Search filter pattern](/patterns/search-filters)).

Reviewer: I think this would read better as three sentences, i.e. full stop after `below`. Also, the double adverb is difficult to say. Do you need `more commonly`? If so, what do you think of `The most common approach is implicitly announcing...`.

Author: Yeah, I wasn't too keen on my wording either

Author: I've just pushed some more wordsmithing

> Avoid using Tags with links and remove buttons together to not confuse users.

Reviewer: The double negative is odd here. Also, should `remove` be `Remove` since that's the name of the button? 

```suggestion
Avoid using Tags with links and Remove buttons together as this will confuse users.
```

		{ label: 'Orange' },
		{ label: 'Peach' },
	]);
	const [announcement, setAnnouncement] = React.useState('');

	const handleRemove = (label) => {
		setItems((prevItems) => prevItems.filter((item) => item.label !== label));
		setAnnouncement(`${label} removed`);
	};

	const itemsWithHandlers = items.map((item) => ({
		...item,
		onRemove: () => handleRemove(item.label),
	}));

	return (
		<>

> 			<Box css={visuallyHiddenStyles} role="status">

Reviewer: Elsewhere we just used a `div` for the hidden status. Given our performance issues with `Box`, I wonder if it's worthwhile maintaining that approach here, even if it's only a minor gain.

Author: had to do this since I can't apply `visuallyHiddenStyles` to a `div` in the code examples

Author: Ultimately, I think we should make VisuallyHidden much more customisable to support role etc, so users can just do `<VisuallyHidden role="status">`
